pkg/ui/src/views/shared/components/sql/box.tsx

```diff
@@ -0,0 +1,32 @@
+import * as hljs from "highlight.js";
+import React from "react";
+
+import "./sqlhighlight.styl";
+
+interface SqlBoxProps {
+  value: string;
+}
+
+export class SqlBox extends React.Component<SqlBoxProps> {
+  preNode: Node;
+
+  shouldComponentUpdate(newProps: SqlBoxProps) {
+    return newProps.value !== this.props.value;
+  }
+
+  componentDidMount() {
+    hljs.highlightBlock(this.preNode);
+  }
+
+  componentDidUpdate() {
+    hljs.highlightBlock(this.preNode);
+  }
+
+  render() {
+    return (
+      <pre className="sql-highlight" ref={ (node) => this.preNode = node }>
```

Comment:
Style nit: I prefer `(node) => { this.preNode = node }`, since having the `=>` and the `=` on the same level (i.e. neither is surrounded by braces) makes it a little hard to read. Don't super care though.